### DISCUSSION TITLE
C - string header issues

### DISCUSSION THREAD

**Discussion by tigercode1118** [microsoft/vscode-cpptools#7442] [Q&A]
After updating my VSCode C/C++ extension, I have been getting squiggles under the function anytime i call the strcpy() function under the <string.h> header. it is saying "too few arguments in invocation of 'strcpy'". I'm not sure if this is something I can fix or if it is a problem with the update. Any ideas?

**Comment by sean-mcmanus** [comment]
Can you provide more info? What version of the string.h header are you using? How many args are you passing and when you do a Go to Definition how many args does the strcpy macro appear to take (I believe it's a macro that may refer to the real function). Also, the output of the C/C++: Log Diagnostics command from the Command Palette could help.

**Reply by tigercode1118** [reply]
I'm on MacOS Big Sur so I'm using the string.h from Apple and the compiler is clang. I was just wondering about this error, because I was never seeing it before, and it is not preventing the code from running. On the integrated terminal it just lists it as a "problem" but the way I am putting it in, as I previously have, is producing the correct outputs. I also ran a previous program that I wrote that used the strcpy() function, which never showed to have any problems before, and it is now producing that problem for every time the function is called. When I scroll over the definition of the function, it is showing "#define strcpy(dest,src,len) __builtin___strcpy_chk(dest, src, XNU_BOS(dest, 1))" but I never looked at it before this started happening so I can't say whether or not that is different than before. Part of me is also wondering if this could be caused by change to the header itself. When I call the function i am entering it as I always have: strcpy(destination string, source string). Also wondering if it's the version of C I am using? Maybe C99 is what I should be using? Here is the output from the Log Diagnostics:

Compiler Path: /usr/bin/clang

Includes:
        /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Kernel.framework/Versions/A/Headers

 /usr/local/include
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.5/include
        /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
    Frameworks:
        /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks

    Standard Version: c11
    IntelliSense Mode: macos-clang-x64
    Other Flags:
        --clang
        --clang_version=100000
Total Memory Usage: 27 MB

**Reply by sean-mcmanus** [reply]
It appears to be picking up the wrong string.h file which has a strcpy that takes 3 arguments. Go to def on `#include <string.h>` should open `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h`. Where is your string.h from? This include "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Kernel.framework/Versions/A/Headers" seems incorrect -- can you remove that?
